components/GoalsForm/GoalsForm.js

```diff
@@ -38,11 +37,41 @@ const GoalSubmitButton = styled.button`
   align-items: center;
 `;
 
-export default function GoalsForm({ onAddGoal }) {
+export default function GoalsForm({
+  onAddGoal,
+  editingGoal,
+  onCancelEdit,
+  savingBalance,
+}) {
   const [isModalOpen, setIsModalOpen] = useState(false);
+  const [formValues, setFormValues] = useState({
+    goalName: "",
+    savedAmount: 0,
+    goalAmount: 0,
+  });
+
+  useEffect(() => {
+    if (editingGoal) {
+      setIsModalOpen(true);
+      setFormValues({
+        goalName: editingGoal.goalName,
+        savedAmount: parseInt(editingGoal.savedAmount),
+        goalAmount: parseInt(editingGoal.goalAmount),
+      });
+    }
+  }, [editingGoal]);
+
+  const handleChange = (event) => {

```

Comment:
to be consistent and because it's nf standard: declare this function with the function keyword